torch/_dynamo/variables/tensor.py

```diff
@@ -681,13 +687,61 @@ def redistribute_fn_with_prim_types(x):
                 mutable_local=variables.base.MutableLocal(),
```

Comment:
right above, here:

```
                fn = fn_var.fn
                name = fn_var.fn.__name__
```

If someone passes in, e.g., a user defined object variable, fn won't be set and you'll trigger a dynamo internal error instead of an intentional graph break. I'd feel better if we exactly match on the class of fn_var here